### PR TITLE
feat(explore): Implement viz switcher redesign

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR implements the redesign of viz switcher. Now user can quickly change viz type to 1 of 6 featured charts without opening the viz gallery modal.
Active chart is represented by an icon and chart name. If chart's name is truncated, then tooltip appears on hover.
The rest of the featured charts are represented only by a clickable icon. Chart's name appears in a tooltip on hover. Clicking the icon switches the viz type.
User can open the viz gallery modal by clicking "View all charts". When user selects a chart that is not "featured", it's represented by a magnifying glass icon and chart's name. It's placed first on the quick viz switcher list.
When a non-featured chart is currently rendered, it's displayed at the start of the list.
Currently rendered chart has a tooltip "Currently rendered: <chart name>" 

Featured charts are:
- Echarts Line Chart
- Table
- Big Number
- Pie Chart
- Echarts Bar Chart
- Echarts Area Chart

Side note - this feature will work even better when Yongjie's [standardized form data PR](https://github.com/apache/superset/pull/20010) is merged  🙂 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/172367517-7458e8a1-3e11-404b-97a7-a541e4ad38d4.mov

### TESTING INSTRUCTIONS
1. Open explore with 1 of the featured charts
2. Verify that there are 6 icons in viz switcher, the active viz is highlighted and its name is visible
3. Click on any other icon, verify that the control panel now displays controls of currently active chart. Click Update Chart and verify that everything works as expected
4. Click "View all charts", change to a non-featured chart
5. Verify that now there are 7 items in viz switcher list and active non-featured chart is 1st on the list. Repeat step 3.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

### CODE REVIEW

**Comment by kasiazjc** [comment]
Looks good to me! 

One thing about the charts: we chose time-series (line, bar) as they will be becoming the default after the generic x-axis will be introduced for everyone. They have also similar statistics in terms of how much they are used (time series vs not). 

Question: do we want to leave time-series charts, go back to simple line/bar OR add a logic which for example shows time-series ones if you have feature flag enabled and for others it shows basic line/bar? All of those are in Tier 1, so I think we need a product decision here. Thoughts @rusackas @alaud?

**Comment by jinghua-qa** [comment]
I like the new feature, the only thing is when i open a non-feature chart, then select the viz switcher, my old chart viz will be gone and i need to go to  "View all charts" to get my old chart viz back but then i have redo everything for my old chart. Can we have a better flow for undo selecting the viz switcher?

https://user-images.githubusercontent.com/81597121/171938467-be4064e7-27f9-4715-a528-462a3be4232b.mov

**Comment by kgabryje** [comment]
> I like the new feature, the only thing is when i open a non-feature chart, then select the viz switcher, my old chart viz will be gone and i need to go to "View all charts" to get my old chart viz back but then i have redo everything for my old chart. Can we have a better flow for undo selecting the viz switcher?

Thanks for testing Jinghua! Now the currently rendered chart's icon stays visible at the start of the list when you switch between charts.
As for preventing the loss of control values when user switches chart, that's a part of Standardized Form Data project, implemented by @zhaoyongjie. Currently only a few charts support the feature of not losing data when viz type changes, but in the following weeks we'll implement it in most charts

File: superset-frontend/src/explore/components/controls/VizTypeControl/FastVizSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { css, SupersetTheme, t, useTheme } from '@superset-ui/core';
+import { usePluginContext } from 'src/components/DynamicPlugins';
+import { Tooltip } from 'src/components/Tooltip';
+import Icons from 'src/components/Icons';
+import { useSelector } from 'react-redux';
+import { ExplorePageState } from '../../../reducers/getInitialState';
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/VizTypeControl/FastVizSwitcher.tsx:32]: nit:
```
import { ExplorePageState } from 'src/explore/reducers/getInitialState';
```

**Review by zhaoyongjie — APPROVED** [review]
Looks cool! I love this feature.

File: superset-frontend/src/explore/components/controls/VizTypeControl/FastVizSwitcher.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/VizTypeControl/FastVizSwitcher.tsx:32]: Oops! I wish auto-import would use those absolute paths by default...

**Comment by lauderbaugh** [comment]
> Looks good to me!
> 
> One thing about the charts: we chose time-series (line, bar) as they will be becoming the default after the generic x-axis will be introduced for everyone. They have also similar statistics in terms of how much they are used (time series vs not).
> 
> Question: do we want to leave time-series charts, go back to simple line/bar OR add a logic which for example shows time-series ones if you have feature flag enabled and for others it shows basic line/bar? All of those are in Tier 1, so I think we need a product decision here. Thoughts @rusackas @lauderbaugh?

**Pull request merged**